lib/solidus_importer/process_row.rb

```diff
@@ -12,12 +12,17 @@ def initialize(importer, row)
     end
 
     def process(initial_context)
-      context = initial_context.dup.merge!(row_id: @row.id, importer: @importer, data: @row.data)
-      @importer.processors.inject(context) do |ctx, processor|
-        next {} if ctx[:success] == false && !processor.ensure_call
-
-        processor.call(ctx)
+      context = initial_context.dup.merge!(importer: @importer, data: @row.data)
+      @importer.processors.each do |processor|
+        begin
+          processor.call(context)
+        rescue => e # rubocop:disable Style/RescueStandardError
+          @importer.process_row_failure(processor, context: context, error: e)
+          context.merge!(success: false, messages: e.message) # rubocop:disable Performance/RedundantMerge
+          break
+        end
       end
+      @importer.logger.info ::SolidusImporter::Log.new(source: @row, context: context)
```

Comment:
Is it possible to pass the source to the logger when initializing it and only send strings as logs?